src/main/java/seedu/address/logic/commands/SuggestCommand.java

```diff
@@ -24,28 +27,45 @@
             + "Example: " + COMMAND_WORD + " " + PREFIX_SUGGEST + SuggestionType.BY_CONTRACT;
     public static final String MESSAGE_SUGGEST_SUCCESS = "Showing clients based on suggestion criteria";
 
-    private final List<Predicate<Client>> suggestionTypePredicateList;
+    private final Set<SuggestionType> suggestionTypeOrderedSet;
+    private final Predicate<Client> suggestionTypePredicate;
+    private final Comparator<Client> suggestionTypeCombinedComparator;
 
     /**
      * Initializes SuggestCommand with a SuggestionType
      */
-    public SuggestCommand(List<Predicate<Client>> suggestionTypePredicateList) {
-        requireNonNull(suggestionTypePredicateList);
-        this.suggestionTypePredicateList = suggestionTypePredicateList;
+    public SuggestCommand(Set<SuggestionType> suggestionTypeOrderedSet) {
+        requireNonNull(suggestionTypeOrderedSet);
+        this.suggestionTypeOrderedSet = suggestionTypeOrderedSet;
+        this.suggestionTypePredicate = suggestionTypeOrderedSet.stream()
+                .map(SuggestionType::getSuggestionPredicate)
+                .reduce(client -> true, (p1, p2) -> client -> p1.test(client) && p2.test(client));
+
+        List<Comparator<Client>> suggestionTypeComparatorList = suggestionTypeOrderedSet.stream()
+                .map(SuggestionType::getSuggestionComparator).collect(Collectors.toList());
+        this.suggestionTypeCombinedComparator = (client1, client2) -> {
```

Comment:
Can this be abstracted to a different method say `generateSuggestionTypeComparatorList` that takes in the `suggestionTypeOrderedSet`?

Reply:
Done, abstracted into a private static method in the same class